### PR TITLE
Created python file to download URL's from server, updated .gitignore…

### PR DESCRIPTION
**Pull request by Joe-TheBro**
…, created Setup.md for dev environment help, and updated README of utils to include information about link-retrieval.py

### CODE REVIEW

File: utils/link-retrieval.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        }
+    }
+    params_json = json.dumps(params)
+
+    # Make the API request and retrieve the data
+    response = requests.get(endpoint, params=params_json)
+    data = response.json()
+
+    # Convert the data to a pandas DataFrame, this feels VERY messy, if you find a better solution, please fix.
+    # Were expecting data that looks like this: [{"page": 0, "totalPages": 100, "totalGameplay": 999999, "gameplay": [{"uuid": "38r8hnf80ew-35uhehrfnjea-wryhghjvsdk", "ytUrl": "https://www.youtube.com/watch?v=dQw4w9WgXcQ"}]},{"page": 0, "totalPages": 100, "totalGameplay": 999999, "gameplay": [{"uuid": "38r8hnf80ew-35uhehrfnjea-wryh32ghjvsdk", "ytUrl": "https://www.youtube.com/watch?v=dQw24w9WgXcQ"}]}]
```
Comment (Why-Ay-Es-Haitch) [inline utils/link-retrieval.py]: I might be wrong but I don't think this is the format in which data would be received. All gameplays from page 0 would be inside `gameplay`. I believe the format would be more like this.
```[{"page": 0, "totalPages": 100, "totalGameplay": 999999, "gameplay": [{"uuid": "38r8hnf80ew-35uhehrfnjea-wryhghjvsdk", "ytUrl": "https://www.youtube.com/watch?v=dQw4w9WgXcQ"}, {"uuid":"38r8hnf80ew-35uhehrfnjea-wryhghjvsdk", "gameplay": "https://www.youtube.com/watch?v=dQw4w9WgXcQ"}]}```
i.e. all gameplays from page 0 would be inside gameplay.

File: Setup.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Development Environment Setup
+- Install Anaconda on server
+- Create a new environment for Waldo under `models/env/`, this has location has already been set in the `.gitignore`
+- Install 
+  - pandas
+  - requests
+  - validators
```
Comment (Why-Ay-Es-Haitch) [inline Setup.md:7]: Do you have a requirements.txt that you can export? Would be alot easier create an environment from that with package versions etc.

File: utils/link-retrieval.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      response_dataframe = pd.concat([response_dataframe, pd.DataFrame(obj[0], index=[0])], ignore_index=True) 
+
+    # Filter out duplicate links
+    response_dataframe.drop_duplicates(subset=["url"], inplace=True)
+
+    # Validate the URLs
+    valid_urls = []
+    for url in response_dataframe["url"]:
+        if (validators.url(url) and is_valid_youtube_url(url)):
+            valid_urls.append(url)
```
Comment (Why-Ay-Es-Haitch) [inline utils/link-retrieval.py:42]: Might be a good idea to move lines 41-54 into a new function which parses the output. Brings more modularity and keeps single responsibility. It'll also allow easier testing down the line.

File: utils/link-retrieval.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+
+    # Make output directory if it doesn't exist
+    download_dir = args['output']
+    if not os.path.exists(download_dir):
+        os.makedirs(download_dir)
+
+    # Save the downloaded links to a file
+    valid_urls_df = pd.DataFrame(valid_urls)
+    valid_urls_df.to_csv((os.path.join(download_dir + "links.csv")), index=True, columns=["uuid","url"])
```
Comment (Why-Ay-Es-Haitch) [inline utils/link-retrieval.py]: Might be a good idea to move all the code in the main body into a separate main method and call that main method with the necessary parameters. This way we can create a pipeline down the track to download urls, download videos, train the model etc.

File: environment.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
+name: /home/overlord/waldo-project/models/env
```
Comment (jaredb1011) [inline environment.yml]: Should we put in a placeholder for the user's path to waldo? 
`<path_to_waldo>/waldo-project/models/env`

If so would also apply to last line of the file

**Review by jaredb1011 — APPROVED** [review]

**Review by Why-Ay-Es-Haitch — APPROVED** [review]
lgtm

**Review by mattolson93 — APPROVED** [review]
looks good

**Pull request merged**